src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -4261,23 +4210,30 @@ private String layoutChars(boolean sci) {
         if (scale == 2  &&
             intCompact >= 0 && intCompact < Integer.MAX_VALUE) {
             // currency fast path
-            int lowInt = (int)intCompact % 100;
             int highInt = (int)intCompact / 100;
-            return (Integer.toString(highInt) + '.' +
-                    StringBuilderHelper.DIGIT_TENS[lowInt] +
-                    StringBuilderHelper.DIGIT_ONES[lowInt]) ;
+            int lowInt = (int) intCompact - highInt * 100;
+            int highIntSize = JLA.stringSize(highInt);
+            byte[] buf = new byte[highIntSize + 3];
+            JLA.getCharsLatin1(highInt, highIntSize, buf);
+            buf[highIntSize] = '.';
+            JLA.writeDigitPairLatin1(buf, highIntSize + 1, lowInt);
+            try {
+                return JLA.newStringNoRepl(buf, StandardCharsets.ISO_8859_1);
+            } catch (CharacterCodingException e) {
+                throw new AssertionError(e);
+            }
         }
 
         StringBuilderHelper sbHelper = new StringBuilderHelper();
-        char[] coeff;
+        byte[] coeff;
         int offset;  // offset is the starting index for coeff array
         // Get the significand as an absolute value
         if (intCompact != INFLATED) {
             offset = sbHelper.putIntCompact(Math.abs(intCompact));
             coeff  = sbHelper.getCompactCharArray();
         } else {
             offset = 0;
-            coeff  = intVal.abs().toString().toCharArray();
```

Comment:
It seems `coeff` is no longer modified after this point. I think you can just use a `String` for `coeff`, and `String` is already coder-optimized, you don't need to expose a StringBuilder.append via JLA.